pkg/gatewayserver/io/grpc/grpc.go

```diff
@@ -108,6 +112,13 @@ func (s *impl) LinkGateway(link ttnpb.GtwGs_LinkGatewayServer) error {
 
 	go func() {
 		for {
+			key := fmt.Sprintf("gs:up:gtw:%s", uid)
+			if limit, _ := s.server.RateLimiter().RateLimit(key, []string{"gs:up"}); limit {
```

Comment:
The resource classes are not prepended to the rate limiting key. This is to avoid confusion in cases where multiple resource classes are set (e.g. for gRPC requests)